prediction_market_agent/agents/autogen_general_agent/social_agent.py

```diff
@@ -0,0 +1,154 @@
+
+from prediction_market_agent.agents.autogen_general_agent.prompts import (
+    CRITIC_PROMPT,
+    INFLUENCER_PROMPT,
+)
+from prediction_market_agent.utils import APIKeys
+
+
+class BetInputPrompt(BaseModel):
+    title: str
+    boolean_outcome: bool
+    collateral_amount: float
+    creation_datetime: datetime
+
+    @staticmethod
+    def from_bet(bet: Bet) -> "BetInputPrompt":
+        return BetInputPrompt(
+            title=bet.market_question,
+            boolean_outcome=bet.outcome,
+            collateral_amount=bet.amount.amount,
+            creation_datetime=bet.created_time,
+        )
+
+
+class AutogenAgentType(str, Enum):
+    WRITER = "writer"
+    CRITIC = "critic"
+    USER = "user"
+
+
+def reflection_message(
+    recipient: UserProxyAgent,
+    messages: list[Dict[Any, Any]],
+    sender: AssistantAgent,
+    config: Optional[Any] = None,
+) -> str:
+    reflect_prompt = Template(CRITIC_PROMPT).substitute(
+        TWEET=recipient.chat_messages_for_summary(sender)[-1]["content"]
+    )
+    return reflect_prompt
+
+
+def build_llm_config(model: str) -> Dict[str, Any]:
+    keys = APIKeys()
+    return {
+        "config_list": [
+            {"model": model, "api_key": keys.openai_api_key.get_secret_value()}
+        ],
+    }
+
+
+def build_agents(model: str) -> Dict[AutogenAgentType, autogen.ConversableAgent]:
+    llm_config = build_llm_config(model)
+
+    user_proxy = autogen.UserProxyAgent(
+        name="User",
+        human_input_mode="NEVER",
+        is_termination_msg=lambda x: x.get("content", "").find("TERMINATE") >= 0,
+        code_execution_config={
+            "last_n_messages": 1,
+            "work_dir": "tasks",
+            "use_docker": False,
+        },
+    )
+
+    writer = autogen.AssistantAgent(
+        name="Writer",
+        llm_config=llm_config,
+        system_message="""
+        You are a professional influencer, known for your insightful and engaging tweets.
+        You transform complex concepts into compelling narratives.
+        You should improve the quality of the content based on the feedback from the user.
+        You must always return only the tweet.
+        """,
+    )
+
+    critic = autogen.AssistantAgent(
+        name="Critic",
+        llm_config=llm_config,
+        system_message="""
+            You are a critic, known for your thoroughness and commitment to standards.
+            Your task is to scrutinize content for any harmful elements or regulatory violations, ensuring
+            all materials align with required guidelines.
+            References to betting and gambling are allowed.
+            """,
+    )
+
+    return {
+        AutogenAgentType.CRITIC: critic,
+        AutogenAgentType.WRITER: writer,
+        AutogenAgentType.USER: user_proxy,
+    }
+
+
+def build_social_media_text(model: str, bets: list[Bet]) -> str | None:
+    """
+    Builds a tweet based on the five markets on Omen that are closing soonest.
```

Comment:
I think this line needs updating. From what I can see there's nothing restricting the number of markets to 5 here. Also, no longer Omen specific.
```suggestion
    Builds a tweet based on a list of bets the user has placed.
```